<launch>
  <arg name="cloud_path" default="$(find point_cloud_segmentation)/test/data/colorized_cloud1.bag"/>
  <arg name="use_pcd" default="false" />
  <arg name="viz" default="true" />

  <!-- Load from a PCD file -->
  <group if="$(arg use_pcd)">
    <remap from="cloud_pcd" to="unmasked_cloud" />
    <node pkg="pcl_ros" type="pcd_to_pointcloud" name="simulated_scanner" output="screen" args="$(arg cloud_path) 1">
      <param name="frame_id" value="base_link"/>
    </node>
  </group>
  <!-- Load from a bag file -->
  <group unless="$(arg use_pcd)">
    <remap from="/camera/depth_registered/points" to="pc_masker_node/unmasked_cloud" />
    <node pkg="rosbag" type="play" name="rosbag_player" args="--clock --loop $(arg cloud_path)"/>
  </group>

  <node pkg="point_cloud_segmentation" type="pc_masker_node" name="pc_masker_node" output="screen" >
    <param name="debug_viewer" value="false" />
    <param name="debug_publisher" value="true" />
  </node>

  <node if="$(arg viz)" name="rviz" type="rviz" pkg="rviz" args="-d $(find point_cloud_segmentation)/config/demo.rviz" output="screen"/>

</launch>
